src/Polly/Policy.ExecuteOverloads.cs

```diff
@@ -10,7 +10,7 @@ public abstract partial class Policy : ISyncPolicy
     /// <param name="action">The action to perform.</param>
     [DebuggerStepThrough]
     public void Execute(Action action) =>
-        Execute((_, _) => action(), new Context(), DefaultCancellationToken);
+        Execute((_, _) => action(), [], DefaultCancellationToken);
```

Comment:
Is this right?  Why [] instead of `new Context()`?

Reply:
The SDK suggests using collection expressions, and in these cases it still works. I can double check that the underlying generated code is the same and it's not a false positive (like the TestData suppressions) but if it's semantically equivalent it's better than fighting the analyzer with suppressions.

Reply:
Just in case, doesn't this creates a new dictionary? I.e. change of behavior and call to a different overload.

Reply:
I'm not sure. I assumed it was some compiler smarts. I'll look at it in ILSpy next time I'm at a computer and see what it's actually producing.

Reply:
Looking at this in Visual Studio shows that it's calling this method:

https://github.com/App-vNext/Polly/blob/a98198f7fc9717babd2ca2f9bb58e65fcfa8c493/src/Polly/Policy.ExecuteOverloads.cs#L60

![image](https://github.com/App-vNext/Polly/assets/1439341/3f417a55-014c-4fd2-bc8a-545cea9db350)

`Context` implements a number of dictionary interfaces, so the compiler is effectively using the collection expression syntax to create an empty `Context`:

https://github.com/App-vNext/Polly/blob/a98198f7fc9717babd2ca2f9bb58e65fcfa8c493/src/Polly/Context.Dictionary.cs#L7

Which agrees with what I can see in ILSpy:

![image](https://github.com/App-vNext/Polly/assets/1439341/9657518c-5ffc-48b7-bbce-25c9a1e7e798)

Reply:
Nice, looks good then!